lib/gui/app/components/drive-selector-button/drive-selector-button.jsx

```diff
@@ -0,0 +1,118 @@
+const middleEllipsis = require('./../../utils/middle-ellipsis')
+
+const { Provider, Txt } = require('rendition')
+const { StepButton, StepNameButton, StepSelection,
+  DetailsText, ChangeButton } = require('./../../styled-components')
+
+class DriveSelectorButton extends React.PureComponent {
+
+  allDevicesFooter() {
+    if (this.props.howManyDeviceSelected > 1) {
+      return this.props.selectedDevices.map((device) =>
+        <Txt key={device.device} tooltip={device.description + '(' + device.displayName + ')'}>
+          { middleEllipsis(device.description, 14) }
+        </Txt>
+       )
+    }
+  }
+
+  render() {
+    if (this.props.hasDrive || !this.props.shouldShowDrivesButton) {
+      return (
+        <Provider>
+          <StepSelection>
+              <StepNameButton
+                plaintext
+                disabled={this.props.disabled}
+                tooltip={this.props.driveListLabel}
+                warning={!this.props.howManyDeviceSelected}
+              >
+                { middleEllipsis(this.props.drivesTitle, 20) }
+                { this.props.hasCompatibilityStatus(this.props.drives(), this.props.image()) ?
+                  <Txt.span className='glyphicon glyphicon-exclamation-sign'
+                    ml='10px'
+                    tooltip={this.props.getCompatibilityStatuses(this.props.drives(),this.props.image())[0].message}
+                  />
+                : null
+                }
+              </StepNameButton>
+
+            <DetailsText>
+              {this.props.driveSize}
+            </DetailsText>
+            { this.props.flashing || !this.props.shouldShowDrivesButton ?
+              null
+              :
+              <ChangeButton
+                plaintext
+                onClick={this.props.reselectDrive}
+              >
+                Change
+              </ChangeButton>
+            }
+            <DetailsText>
+              {this.allDevicesFooter()}
+            </DetailsText>
+          </StepSelection>
+        </Provider>
+      )
+    }
+    else {
+      return (
+        <Provider>
+          <StepSelection>
+            <StepButton
+              primary
+              disabled={this.props.disabled}
+              onClick={this.props.openDriveSelector}
+            >
+              Select drive react
+            </StepButton>
+          </StepSelection>
+        </Provider>
+      )
+    }
+  }
+}
+
+DriveSelectorButton.propTypes = {
+  hasDrive: propTypes.bool,
+  shouldShowDrivesButton: propTypes.bool,
+  disabled: propTypes.bool,
+  drivesTitle: propTypes.string,
+  driveListLabel: propTypes.string,
+  openDriveSelector: propTypes.func,
+  howManyDeviceSelected: propTypes.number,
+  reselectDrive: propTypes.func,
+  driveSize: propTypes.string,
+  hasCompatibilityStatus: propTypes.func,
+  getCompatibilityStatuses: propTypes.func,
+  drives: propTypes.func,
+  image: propTypes.func,
+  selectedDevices: propTypes.array,
+}
+
+module.exports = DriveSelectorButton
```

Comment:
Can you use a named export here?

Reply:
I cannot as long as we need to use React2Angular component - I haven't found the way for it to accept the named export. Once the whole main page will be converted to React this is going to be changed.

Reply:
@amdomanska How are you importing this module? modules and `require` statements should have nothing to do with `react2angular`.

Reply:
Tests fail for some reason when we use named exports, I still need to take a closer look at this